src/transformStateWithClones.js

```diff
@@ -7,7 +7,36 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  let stateCopy = { ...state };
+  let tempObj = {};
+  let stateArray = [];
+
+  for (let i = 0; i < actions.length; i++) {
+    switch (actions[i].type) {
+      case 'addProperties':
+        stateCopy = {
+          ...stateCopy,
+          ...actions[i].extraData,
+        };
+        tempObj = { ...stateCopy };
+        stateArray = [...stateArray, tempObj];
+        break;
+
+      case 'removeProperties':
+        for (const item of actions[i].keysToRemove) {
+          delete stateCopy[item];
+        }
+        stateArray = [...stateArray, stateCopy];
+        break;
+
+      case 'clear':
+        stateCopy = {};
+        stateArray = [...stateArray, stateCopy];
```

Comment:
`stateArray = [...stateArray, stateCopy];` is duplicated three times, so make sense to move it after the switch case block